src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaLineItemService.java

```diff
@@ -82,4 +86,34 @@ public LineItem createDuplicate(LineItem originalLineItem, BudgetScenario budget
     public void deleteMany(List<Integer> lineItemIds) {
         this.lineItemRepository.deleteLineItemsWithIds(lineItemIds);
     }
+
+    @Override
+    public void createLineItemFromTeachingAssignmentAndBudgetScenario(TeachingAssignment teachingAssignment, BudgetScenario budgetScenario) {
+        if (teachingAssignment.isApproved() == false) {
+            return;
+        }
+
+        LineItem lineItemDTO = null;
+
+        if (teachingAssignment.isBuyout()) {
+            lineItemDTO = new LineItem();
+            lineItemDTO.setAmount(0f);
+            lineItemDTO.setLineItemCategory(lineItemCategoryService.findById(2));
+            lineItemDTO.setBudgetScenario(budgetScenario);
+
+            String description = teachingAssignment.getInstructor().getFullName() + " Buyout Funds for " + Term.getRegistrarName(teachingAssignment.getTermCode());
+            lineItemDTO.setDescription(description);
+            this.findOrCreate(lineItemDTO);
```

Comment:
this.findOrCreate(lineItemDTO) can be taken out of both cases and placed after the 'if' statement.

Reply:
Actually, a number of these lines are the same. Please move the non-unique lines out.